### PR TITLE
python311Packages.yaspin: fix build, minor cleanup, python311Packages.termcolor: update meta

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

see commits

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/yaspin/default.nix [hunk 1]
```diff
@@ -30,14 +31,16 @@ buildPythonPackage rec {
...
 
-  postPatch = ''
-    # https://github.com/pavdmyt/yaspin/pull/212
-    substituteInPlace pyproject.toml \
-      --replace 'termcolor-whl = "1.1.2"' 'termcolor = "*"'
-  '';
+  disabledTestPaths = [
+    # AssertionError: assert False
+    #  +  where False = <built-in method startswith of str object at 0x7ffff3f0b330>('\r\x1b')
+    #  +    where <built-in method startswith of str object at 0x7ffff3f0b330> = '\r/ '.startswith
```
Comment (samuela) [inline pkgs/development/python-modules/yaspin/default.nix:41]: let's open an issue upstream and link to it in the comment here

Comment (SuperSandro2000) [reply pkgs/development/python-modules/yaspin/default.nix:41]: No need to do that, I figured it out on my own. We are using the dumb TERM in the builder and termcolor deactivates colored output with that. \x1b is \033 which is the standard unix color escape code used in colors.

Comment (samuela) [reply pkgs/development/python-modules/yaspin/default.nix]: ohhh ok nice find -- let's leave a comment to that effect then

Comment (SuperSandro2000) [reply pkgs/development/python-modules/yaspin/default.nix]: Already added that. Otherwise we will forget pretty fast why we set that env. 😂

File: pkgs/development/python-modules/yaspin/default.nix [hunk 2]
```diff
@@ -30,13 +31,13 @@ buildPythonPackage rec {
...
+    pytest-xdist
     pytestCheckHook
   ];
 
-  postPatch = ''
-    # https://github.com/pavdmyt/yaspin/pull/212
-    substituteInPlace pyproject.toml \
-      --replace 'termcolor-whl = "1.1.2"' 'termcolor = "*"'
+  preCheck = ''
+    # tests assert for \033 which gets turned off in termcolor when TERM dumb is used which is used by nix
```
Comment (samuela) [inline pkgs/development/python-modules/yaspin/default.nix:39]: let's avoid putting comments in nix strings. they unnecessarily affect drv hashes

File: pkgs/development/python-modules/termcolor/default.nix
Comment (samuela) [inline pkgs/development/python-modules/termcolor/default.nix:1]: let's make this change in a separate PR

**Review by samuela — DISMISSED** [review]
nice find with FORCE_COLOR, just two comments

File: pkgs/development/python-modules/termcolor/default.nix
Comment (SuperSandro2000) [reply pkgs/development/python-modules/termcolor/default.nix:1]: Come on, this really doesn't matter.

File: pkgs/development/python-modules/yaspin/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/yaspin/default.nix:39]: If you want to but that also doesn't really matter, not like we are going to update it anytime soon.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281299` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.yaspin</li>
    <li>python311Packages.yaspin.dist</li>
    <li>python312Packages.yaspin</li>
    <li>python312Packages.yaspin.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Change looks good to me.
